UnsatisfiedLinkError issue on first build (React Native v0.69)

<details>
<summary>See the error report</summary>

```
FATAL EXCEPTION: create_react_context

Caused by: java.lang.UnsatisfiedLinkError: dlopen failed: library "libv8android.so" not found: needed by /data/app/~~i0I1NaaZCLMDKne_2BVLeQ==/com.testapp-f9Qoatevq6ZoTI1QzWzlnw==/lib/arm64/libv8executor.so in namespace classloader-namespace

FATAL EXCEPTION: create_react_context
Process: com.testapp, PID: 4353
java.lang.UnsatisfiedLinkError: couldn't find DSO to load: libv8executor.so caused by: dlopen failed: library "libv8android.so" not found: needed by /data/app/~~i0I1NaaZCLMDKne_2BVLeQ==/com.testapp-f9Qoatevq6ZoTI1QzWzlnw==/lib/arm64/libv8executor.so in namespace classloader-namespace result: 0
	at com.facebook.soloader.SoLoader.doLoadLibraryBySoName(SoLoader.java:1098)
	at com.facebook.soloader.SoLoader.loadLibraryBySoNameImpl(SoLoader.java:914)
	at com.facebook.soloader.SoLoader.loadLibraryBySoName(SoLoader.java:826)
	at com.facebook.soloader.SoLoader.loadLibrary(SoLoader.java:773)
	at com.facebook.soloader.SoLoader.loadLibrary(SoLoader.java:743)
	at io.csie.kudo.reactnative.v8.executor.V8Executor.<clinit>(V8Executor.java:22)
	at io.csie.kudo.reactnative.v8.executor.V8ExecutorFactory.create(V8ExecutorFactory.java:50)
	at com.facebook.react.ReactInstanceManager$5.run(ReactInstanceManager.java:1109)
	at java.lang.Thread.run(Thread.java:920)
Caused by: java.lang.UnsatisfiedLinkError: dlopen failed: library "libv8android.so" not found: needed by /data/app/~~i0I1NaaZCLMDKne_2BVLeQ==/com.testapp-f9Qoatevq6ZoTI1QzWzlnw==/lib/arm64/libv8executor.so in namespace classloader-namespace
	at java.lang.Runtime.load0(Runtime.java:929)
	at java.lang.System.load(System.java:1620)
	at com.facebook.soloader.SoLoader$1.load(SoLoader.java:529)
	at com.facebook.soloader.DirectorySoSource.loadLibraryFrom(DirectorySoSource.java:110)
	at com.facebook.soloader.DirectorySoSource.loadLibrary(DirectorySoSource.java:63)
	at com.facebook.soloader.ApplicationSoSource.loadLibrary(ApplicationSoSource.java:91)
	at com.facebook.soloader.SoLoader.doLoadLibraryBySoName(SoLoader.java:1038)
	... 8 more
```
</details>

The first build doesn't seem to include the DSO files of react-native-v8, I see that the apk size is incorrect:

- First build or `./gradlew clean` + rebuild: apk size is 45.6MB
- Second build: apk size is 85.1MB and it works fine

I've create a example project for reproduce the issue, it just an initial project and followed [react-native-v8 setup](https://github.com/Kudo/react-native-v8#installation-for-react-native--066): https://github.com/jhen0409/react-native-v8-issue-on-0.69

My environment is macOS with M1 Max chip.

I got this issue when I try to upgrade from RN 0.68 to 0.69.

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -218,7 +218,7 @@ android {
   }
   sourceSets {
     main {
-      jniLibs.srcDirs = ["${extractSoDir}/v8/jni"]
+      jniLibs.srcDirs += ["${extractSoDir}/v8/jni"]
 
       if (v8UseSnapshot) {
         assets.srcDirs += [ "${v8AndroidDir}/dist/snapshot_blob" ]
@@ -434,21 +434,31 @@ task downloadNdkBuildDependencies {
 task prepareThirdPartyNdkHeaders(dependsOn:[downloadNdkBuildDependencies, prepareBoost, prepareDoubleConversion, prepareFolly, prepareGlog]) {
 }
 
-tasks.whenTaskAdded { task ->
-  def buildType = task.name.endsWith('Debug') ? 'Debug' : 'Release'
-  if (
-      !task.name.contains("Clean") && (
-        task.name.startsWith("externalNativeBuild")
-        || task.name.startsWith("buildCMake")
-        || task.name.startsWith("configureCMake"))
-  ) {
-    task.dependsOn(prepareThirdPartyNdkHeaders)
-    extractAARHeaders.dependsOn(prepareThirdPartyNdkHeaders)
-    extractSOFiles.dependsOn(prepareThirdPartyNdkHeaders)
-    task.dependsOn(extractAARHeaders)
-    task.dependsOn(extractSOFiles)
-    task.dependsOn("extractReactNativeAAR${buildType}")
-  } else if (task.name == "merge${buildType}JniLibFolders") {
-    task.dependsOn(extractSOFiles)
+def nativeBuildDependsOn(dependsOnTask, buildTypesIncludes) {
+  def buildTasks = tasks.findAll { task ->
+    def taskName = task.name
+    if (taskName.contains("Clean")) { return false }
+    if (taskName.contains("externalNative") || taskName.contains("CMake") || taskName.contains("generateJsonModel")) {
+      if (buildTypesIncludes == null) { return true }
+      for (buildType in buildTypesIncludes) {
+        if (taskName.contains(buildType)) { return true }
+      }
+    }
+    return false
   }
+  buildTasks.forEach { task -> task.dependsOn(dependsOnTask) }
+}
+
+afterEvaluate {
+  extractAARHeaders.dependsOn(prepareThirdPartyNdkHeaders)
+  extractSOFiles.dependsOn(prepareThirdPartyNdkHeaders)
+
+  nativeBuildDependsOn(prepareThirdPartyNdkHeaders, null)
+  nativeBuildDependsOn(extractAARHeaders, null)
+  nativeBuildDependsOn(extractSOFiles, null)
+  nativeBuildDependsOn(extractReactNativeAARRelease, ["Release", "RelWithDebInfo"])
+  nativeBuildDependsOn(extractReactNativeAARDebug, ["Debug"])
+
+  tasks.findByName("mergeReleaseJniLibFolders")?.dependsOn(extractSOFiles)
+  tasks.findByName("mergeDebugJniLibFolders")?.dependsOn(extractSOFiles)
 }

